database/ticket.go

```diff
@@ -82,6 +82,13 @@ var (
 )
 
 func (vdb *VspDatabase) InsertNewTicket(ticket Ticket) error {
+	// Marshal ticket first. This could prevent a DB transaction being started
+	// and rolled back unnecessarily.
+	ticketBytes, err := json.Marshal(ticket)
```

Comment:
Hmm, could be said that the ticket is also being marshalled too early looking at where its used. Also isn't it highly unlikely this the marshal fails? If that's the case then the possibility of a rollback would be minimal.

Reply:
A place I think we could look for some performance gains is iterating the ticket bucket to check if there is a ticket with the same fee address. My thinking is if the fee address and index are part of the key of the ticket then it can be extracted from it and used in the comparison and there would be no need to unmarshal the ticket value just for the comparison which should speed things up. Thoughts?

Reply:
The idea to change the DB keys would help to iterate over the tickets quicker, but would require a DB migration and is too much to change in this PR. We can investigate that idea later if the time to iterate becomes an issue in future.